src/NServiceBus.Core/Transports/ErrorHandleResult.cs

```diff
@@ -0,0 +1,18 @@
+﻿namespace NServiceBus.Transport
+{
+    /// <summary>
+    /// Provides information about error handling.
+    /// </summary>
+    public enum ErrorHandleResult
+    {
+        /// <summary>
+        /// Indicates that the infrastructure handled the current error.
+        /// </summary>
+        Handled,
+
+        /// <summary>
+        /// Indicates that the infrastructure was did not handle the current error. A retry is necessary.
+        /// </summary>
+        RetryNecessary
```

Comment:
should that rather be `RetryRequired`? Based on this:

> "Required" is used in the context of arbitrary rules and regulations, that is, when there is some requirement by law or convention that must be met in order for something to happen.
https://www.quora.com/What-is-the-difference-between-required-and-necessary